rust/src/lib/ifaces/inter_ifaces.rs

```diff
@@ -411,6 +421,51 @@ impl Interfaces {
                     break;
                 }
             }
+            // If `permanent_mac_address` got no matches, fallback to
+            // `mac_address`
+            if !has_match {
```

Comment:
This block is basically duplicating the same operations done above. Maybe we can improve it with a loop?

For example: 
```
for use_permanent in &[true, false] {
    let addr = if use_permanent {
        cur_iface.base_iface().permanent_mac_address
    } else {
        cur_iface.base_iface().mac_address
    }
    if addr
        .as_deref()
         .map(|m| m.to_ascii_uppercase())
         .as_deref()
         == Some(&mac_address)
     { ... }
```

Reply:
But the logic seems to be wrong here, we do not know if we should use the `use_permanent` address beforehand until we can not find a match when iterating through the interfaces, then we fallback to mac address.

Reply:
We need to search all permanent mac address first, then fallback to active mac address.

Reply:
My bad. Yes. It is doable.